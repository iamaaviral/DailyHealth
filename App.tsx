import * as React from "react";
import { Component } from "react";
import { Platform, StyleSheet, Text, View } from "react-native";
import Bmi from "./components/bmi";
import { Tabs } from "./components/config/router";
import Header from "./components/Header";

// const word: any = (data).name;

const instructions = Platform.select({
  android:
    "Double tap R on your keyboard to reload,\n" +
    "Shake or press menu button for dev menu",
  ios: "Press Cmd+R to reload,\n" + "Cmd+D or shake for dev menu",
});

export default class App extends Component<any, any> {
  constructor(props: any) {
    super(props);
    this.state = {
      data: {
        calories: 37,
        climbed: 0,
        distance: 1050,
        heartRate: 72,
        height: 1.7,
        steps: 1505,
        weight: 64,
       },
       editing: false,
       value: 45,
    };
  }

  onSubmit = (data: any) => {
    if (this.state.editing) {
      // this.setState({
      //   value: data,
      //   editing : !this.state.editing,
      // });
    this.setState((prevState: any) => ({
    data: {
      ...prevState.data,
      weight: data,
  },
  editing : !this.state.editing,
}));
} else {
      this.setState({
        editing : !this.state.editing,
      });
    }
  }
  public render() {
    return (
       <View style={styles.container}>
              <Header />
              <View style={styles.body}>
                <Tabs
                  screenProps={{
                    data: this.state.data,
                    editing: this.state.editing,
                    // handleClick() {this.onSubmit.bind(this); },
                    handlerFromParent: this.onSubmit.bind(this),
                  }}
                />
              </View>
              <Bmi
                data={this.state.data}
                handleClick={this.onSubmit.bind(this)}
              />
            </View>
    );
  }
}

const styles = StyleSheet.create({
  body: {
    // flex: 25,
    // justifyContent: 'space-between',
    alignSelf: "stretch",
    flex: 15,
    flexDirection: "column",
  },
  container: {
    alignItems: "center",
    alignSelf: "stretch",
    flex: 1,
    justifyContent: "center",
  },
});
