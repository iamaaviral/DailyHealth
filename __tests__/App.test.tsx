// components/__tests__/Hello.tsx
import React from "react";
import "react-native";
import renderer from "react-test-renderer";

import App from "../App";

test("App snapshot", () => {
  const snap = renderer.create(
    <App />,
  ).toJSON();
  expect(snap).toMatchSnapshot();
});
