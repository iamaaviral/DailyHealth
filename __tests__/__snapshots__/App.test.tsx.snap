// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "rgba(0,170,170,0.1)",
        "flex": 1,
        "flexDirection": "column",
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 25,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Daily health
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 15,
        "flexDirection": "column",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "alignItems": "stretch",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              removeClippedSubviews={false}
              style={
                Object {
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "padding": 15,
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "column",
                          "justifyContent": "space-around",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <Image
                            source={
                              Object {
                                "testUri": "../../../assets/weight.png",
                              }
                            }
                            style={
                              Object {
                                "height": 15,
                                "width": 15,
                              }
                            }
                          />
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "fontSize": 17,
                                "paddingLeft": 10,
                              }
                            }
                          >
                            Weight
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <Image
                            source={
                              Object {
                                "testUri": "../../../assets/height.png",
                              }
                            }
                            style={
                              Object {
                                "height": 15,
                                "width": 15,
                              }
                            }
                          />
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "fontSize": 17,
                                "paddingLeft": 10,
                              }
                            }
                          >
                            Height
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <Image
                            source={
                              Object {
                                "testUri": "../../../assets/steps.png",
                              }
                            }
                            style={
                              Object {
                                "height": 15,
                                "width": 15,
                              }
                            }
                          />
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "fontSize": 17,
                                "paddingLeft": 10,
                              }
                            }
                          >
                            Steps
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <Image
                            source={
                              Object {
                                "testUri": "../../../assets/distance.png",
                              }
                            }
                            style={
                              Object {
                                "height": 15,
                                "width": 15,
                              }
                            }
                          />
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "fontSize": 17,
                                "paddingLeft": 10,
                              }
                            }
                          >
                            Distance
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <Image
                            source={
                              Object {
                                "testUri": "../../../assets/climb.png",
                              }
                            }
                            style={
                              Object {
                                "height": 15,
                                "width": 15,
                              }
                            }
                          />
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "fontSize": 17,
                                "paddingLeft": 10,
                              }
                            }
                          >
                            Climbed
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <Image
                            source={
                              Object {
                                "testUri": "../../../assets/calories.png",
                              }
                            }
                            style={
                              Object {
                                "height": 15,
                                "width": 15,
                              }
                            }
                          />
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "fontSize": 17,
                                "paddingLeft": 10,
                              }
                            }
                          >
                            Calories
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <Image
                            source={
                              Object {
                                "testUri": "../../../assets/heart.png",
                              }
                            }
                            style={
                              Object {
                                "height": 15,
                                "width": 15,
                              }
                            }
                          />
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "fontSize": 17,
                                "paddingLeft": 10,
                              }
                            }
                          >
                            Heart rate
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "column",
                          "justifyContent": "space-around",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "fontSize": 17,
                              "paddingLeft": 10,
                            }
                          }
                        >
                          64
                           kg
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "fontSize": 17,
                              "paddingLeft": 10,
                            }
                          }
                        >
                          1.7
                           m
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "fontSize": 17,
                              "paddingLeft": 10,
                            }
                          }
                        >
                          1505
                           steps
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "fontSize": 17,
                              "paddingLeft": 10,
                            }
                          }
                        >
                          1050
                           m
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "fontSize": 17,
                              "paddingLeft": 10,
                            }
                          }
                        >
                          0
                           m
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "fontSize": 17,
                              "paddingLeft": 10,
                            }
                          }
                        >
                          0.037
                           kcal
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "fontSize": 17,
                              "paddingLeft": 10,
                            }
                          }
                        >
                          72
                           bps
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={null}
          />
        </View>
      </View>
      <View>
        <View
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "backgroundColor": "#F7F7F7",
              "borderTopColor": "rgba(0, 0, 0, .3)",
              "borderTopWidth": 0.5,
              "flexDirection": "row",
              "height": 49,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            }
          }
        >
          <View
            accessible={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0, 0, 0, 0)",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                  },
                  Object {
                    "flexDirection": "column",
                    "justifyContent": "flex-end",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "height": 29,
                    },
                    false,
                    Object {
                      "flex": 1,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 30,
                      "opacity": 1,
                      "position": "absolute",
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        false,
                        false,
                        false,
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                    underlayColor="white"
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#3478f6",
                            "fontSize": 35,
                          },
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                            },
                            undefined,
                          ],
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 30,
                      "opacity": 0,
                      "position": "absolute",
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        false,
                        false,
                        false,
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                    underlayColor="white"
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#929292",
                            "fontSize": 35,
                          },
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                            },
                            undefined,
                          ],
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(52, 120, 246, 1)",
                    "fontSize": 10,
                    "marginBottom": 1.5,
                    "textAlign": "center",
                  }
                }
              >
                in MKS
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0, 0, 0, 0)",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                  },
                  Object {
                    "flexDirection": "column",
                    "justifyContent": "flex-end",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "height": 29,
                    },
                    false,
                    Object {
                      "flex": 1,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 30,
                      "opacity": 0,
                      "position": "absolute",
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        false,
                        false,
                        false,
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                    underlayColor="white"
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#3478f6",
                            "fontSize": 35,
                          },
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                            },
                            undefined,
                          ],
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 30,
                      "opacity": 1,
                      "position": "absolute",
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        false,
                        false,
                        false,
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                    underlayColor="white"
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#929292",
                            "fontSize": 35,
                          },
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                            },
                            undefined,
                          ],
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(146, 146, 146, 1)",
                    "fontSize": 10,
                    "marginBottom": 1.5,
                    "textAlign": "center",
                  }
                }
              >
                in CGS
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,170,170,0.1)",
        "flex": 2,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Your BMI : 
      22.15
    </Text>
    <View
      style={
        Object {
          "flex": 1,
          "paddingLeft": 125,
          "paddingRight": 20,
          "paddingVertical": 0,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel="decrement"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "blue",
                },
              ]
            }
          >
            edit
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
